htlcswitch/switch.go

```diff
@@ -364,9 +367,15 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, paymentID uint64,
 		return zeroPreimage, err
 	}
 
+	// To ensure atomicity between checking the PreimageCache and
+	// forwarding the HTLC (committing the circuit), we aquire a multimutex
+	// for this paymentID.
+	s.paymentIDMtx.Lock(paymentID)
```

Comment:
Can the multi mutex be avoided by looking up the preimage after commiting the circuit? If it is present then, the circuit could be teared down again. Or left in half open state, where it will be cleaned up later.

Reply:
Very good idea, don!